libp2p/protocol_muxer/multiselect.py

```diff
@@ -12,18 +18,20 @@ class Multiselect(IMultiselectMuxer):
     a specific protocol and handler pair to use for communication
     """
 
-    def __init__(self):
+    handlers: Dict[TProtocol, StreamHandlerFn]
+
+    def __init__(self) -> None:
         self.handlers = {}
 
-    def add_handler(self, protocol, handler):
+    def add_handler(self, protocol: TProtocol, handler: StreamHandlerFn) -> None:
         """
         Store the handler with the given protocol
         :param protocol: protocol name
         :param handler: handler function
         """
         self.handlers[protocol] = handler
 
-    async def negotiate(self, stream):
+    async def negotiate(self, stream: IMuxedStream) -> Tuple[TProtocol, StreamHandlerFn]:
```

Comment:
```suggestion
    async def negotiate(self, stream: Union[IMuxedStream, IRawConnection]) -> Tuple[TProtocol, StreamHandlerFn]:
```
From the code snippet below, we know that `Multiselect` and `MultiselectClient` are used not only for the streams but connections for selecting protocol for connections(security, and multiplexer). Therefore, the type should include `IRawConnection`. I'm thinking we can add some type like `NegotiableTransport = Union[IMuxedStream, IRawConnection]`.
https://github.com/libp2p/py-libp2p/blob/7477b295082e47f92ae7f03398b4cdb4aef9b248/libp2p/security/security_multistream.py#L94-L99